gammapy/makers/utils.py

```diff
@@ -422,3 +424,161 @@ def make_theta_squared_table(
     table.meta["ON_RA"] = position.icrs.ra
     table.meta["ON_DEC"] = position.icrs.dec
     return table
+
+
+def get_rad_max_vs_energy(geom, observation):
+    """Obtain the values of `RAD_MAX` values corresponding to the observation
+    offset and to the bin centers of the energy axis in the `RegionGeom`.
+
+    Parameters
+    ----------
+    geom : `~gammapy.maps.RegionGeom`
+        Reference map geom.
+    observation : `~gammapy.data.Observation`
+        Observation container.
+
+    Returns
+    -------
+    array : `~astropy.units.Quantity`
+        Values of the `RAD_MAX` corresponding to each estimated energy bin center.
+    """
+    if observation.rad_max is None:
+        raise ValueError(
+            "the IRF for this observation does not include a RAD_MAX_2D table"
+        )
+
+    on_center = geom.center_skydir
+    offset = on_center.separation(observation.pointing_radec)
+
+    rad_max = observation.rad_max.evaluate(
+        offset=offset, energy=geom.axes["energy"].center
+    )
+
+    return rad_max
+
+
+def make_counts_rad_max(geom, observation):
```

Comment:
One last minor comment here: we typically kept the signature of the `make_something` utility methods to take the IRF classes and not the observation objects. The idea I just that this way they can used a bit easier independently. The API relying on the `Obseervation` object is still available on the `Maker` class, like `MapDatasetMaker.make_off_counts(geom, observation)`. So maybe changee the signature here to take the `geom`, `rad_max` and `events` independently?

Reply:
Hello @adonath, I changed the function's signature from `make_counts_rad_max(geom, observation)` to `make_counts_rad_max(geom, rad_max, events)`.